File: QuickCar/Core/Export/PDFReportBuilder.swift
import UIKit

enum PDFReportBuilder {
    static func build(trips: [Trip], settings: AppSettings, vehicleName: String) -> URL? {
        let page = CGRect(x: 0, y: 0, width: 595, height: 842)
        let renderer = UIGraphicsPDFRenderer(bounds: page)
        let url = FileManager.default.temporaryDirectory
            .appendingPathComponent("quickcar-monthly-\(Int(Date().timeIntervalSince1970)).pdf")
        let summary = DrivingSummary(trips: trips, pricePerLiter: settings.pricePerLiter)
        do {
            try renderer.writePDF(to: url) { ctx in
                ctx.beginPage()
                let title = "QuickCar — \(vehicleName)"
                (title as NSString).draw(at: CGPoint(x: 40, y: 40), withAttributes: [
                    .font: UIFont.boldSystemFont(ofSize: 18)
                ])
                var y: CGFloat = 80
                let lines = [
                    "Trips: \(summary.tripCount)",
                    "Distance: \(String(format: "%.1f", summary.distanceKm)) km",
                    "Fuel: \(String(format: "%.2f", summary.fuelUsedL)) L",
                    "Avg: \(String(format: "%.1f", summary.avgL100)) L/100km",
                    "Cost: \(String(format: "%.2f", summary.estimatedCost))",
                    "Score: \(summary.avgScore.map { String(format: "%.0f", $0) } ?? "—")"
                ]
                for line in lines {
                    (line as NSString).draw(at: CGPoint(x: 40, y: y), withAttributes: [
                        .font: UIFont.systemFont(ofSize: 12)
                    ])
                    y += 22
                }
                ("QuickCar" as NSString).draw(
                    at: CGPoint(x: 40, y: 800),
                    withAttributes: [.font: UIFont.systemFont(ofSize: 10)]
                )
            }
            return url
        } catch {
            Log.error("PDF failed: \(error)")
            return nil
        }
    }
}

File: mobile/src/core/export/pdfReportBuilder.ts
import * as Print from "expo-print";
import { summarize, emptyDrivingSummary } from "../storage/models";
import type { Trip } from "../storage/models";

export async function buildMonthlyReportPDF(
  trips: Trip[],
  pricePerLiter: number,
  vehicleName: string
): Promise<string | undefined> {
  const summary = trips.length ? summarize(trips, pricePerLiter) : emptyDrivingSummary();
  const html = `
    <html>
      <body style="font-family: -apple-system, Helvetica, sans-serif; padding: 40px;">
        <h1 style="font-size: 20px;">QuickCar — ${vehicleName || "Vehicle"}</h1>
        <p style="font-size: 13px; line-height: 1.8;">
          Trips: ${summary.tripCount}<br/>
          Distance: ${summary.distanceKm.toFixed(1)} km<br/>
          Fuel: ${summary.fuelUsedL.toFixed(2)} L<br/>
          Avg: ${summary.avgL100.toFixed(1)} L/100km<br/>
          Cost: ${summary.estimatedCost.toFixed(2)}<br/>
          Score: ${summary.avgScore != null ? summary.avgScore.toFixed(0) : "—"}
        </p>
        <p style="font-size: 10px; color: #888;">QuickCar</p>
      </body>
    </html>
  `;
  try {
    const { uri } = await Print.printToFileAsync({ html });
    return uri;
  } catch {
    return undefined;
  }
}
